pkg/agent/handler/managedapplicationprofile.go

```diff
@@ -117,6 +118,40 @@ func (h *managedApplicationProfile) onPending(ctx context.Context, profile *mana
 	return err
 }
 
+func (h *managedApplicationProfile) checkForEnumValueMap(_ context.Context, data map[string]interface{}, profileDef string) {
+	log := h.logger.WithField("applicationProfileDefinition", profileDef)
+
+	// get application profile definition
+	ri, err := h.cache.GetApplicationProfileDefinitionByName(profileDef)
+	if err != nil {
+		log.WithError(err).Error("error getting application profile definition from cache")
+		return
+	}
+	if ri == nil {
+		log.Debug("could not find application profile definition in cache")
+		return
+	}
+	appProfDef := management.ApplicationProfileDefinition{}
+	if err := appProfDef.FromInstance(ri); err != nil {
+		log.WithError(err).Error("error reading application profile definition from cache")
+		return
+	}
+
+	updateDataFromEnumMap(data, appProfDef.Spec.Schema)
+}
+
+func (h *managedApplicationProfile) getManagedApp(_ context.Context, profile *management.ManagedApplicationProfile) (*management.ManagedApplication, error) {
+	app := management.NewManagedApplication(profile.Spec.ManagedApplication, profile.Metadata.Scope.Name)
+	ri, err := h.client.GetResource(app.GetSelfLink())
```

Comment:
Do we have to get the app from server or can we get it from cache?

Reply:
I will give that a try, I had that there but hit an issue, will give it another go. Will change for creds and acc reqs too.

Reply:
DAs do not cache managed applications. We should consider adding that, but in another story?